File: 2749/Solution.java
class Solution {
    public int makeTheIntegerZero(int num1, int num2) {
        long n1 = num1;
        long n2 = num2;
        for (int k = 1; k <= 100; k++) {
            long x = n1 - k * n2;
            if (x < k) {
                continue;
            }
            int bits = Long.bitCount(x);
            if (bits <= k) {
                return k;
            }
        }
        return -1;        
    }
}

File: 2749/Solution.py
class Solution(object):
    def makeTheIntegerZero(self, num1, num2):
        n1 = num1
        n2 = num2
        for k in range(1, 101):
            x = n1 - k * n2
            if x < k:
                continue
            if bin(x).count('1') <= k:
                return k
        return -1